### PR TITLE
Meta SNIP-25: security council

### PR DESCRIPTION
**Pull request by Eikix**
Drafting the meta SNIP for Security Council as part of the [Fête du SNIP](https://lu.ma/LeFeteduSNIP?locale=fr&tk=vmF03x)

### CODE REVIEW

**Comment by lucadonnoh** [comment]
For reference, this is the current upgradability structure:

image

Starknet would need to give up the permissions for all Proxy governors, Implementation governors, Verifier governors and bridge owners. Also, as per the [Risk rosette framework](https://gov.l2beat.com/t/the-risk-rosette-framework/292) and the Stages framework, the Exit window and Stage designation will stay at zero until forced transactions are implemented.

File: SNIPS/snip-75.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+- Election and Composition of the council
+
+## Examples and hypothetical future
+
+As a bi-product of the Security Council creation, all upgrades on the Starknet network would be delayed by an amount of time chosen by the Security Council, e.g. 1 month.
+Simple minimal examples of how the Security Council could look like (inspired the current states of other rollups):
+
+- Emergency Actions: the Security Council can reduce the upgrade delay time to zero for an emergency.
+- Non-Emergency Action: the Security Council can vote to accept an upgrade proposed by Starkware.
+- Composition: the Security Council is composed of 12 to 15 people from broad set of sectors (Builders, Users, DeFi, Gaming, Wallet, Cryptography, Infra, etc.), covering a large amount of timezones (always 75% of the council is susceptible to be awake at all time "The Sun never sets on the Security Council" 👑 👀). Each person in the council has at least 1M vSTRK delegated to them (they must have either skin in the game or a "vote of confidence" from token holders). Council members are reelected by the Starknet Foundation and can be removed by their peers should they misbehave.
```
[the remark this answers is not in the document]
Comment (Eikix) [reply SNIPS/snip-75.md:56]: I meant:
If the rollup contract is controlled by SW multisig but has timelock
And SC multisig is controlled by its members and has special right on the rollup contract

[the remark this answers is not in the document]
Comment (lucadonnoh) [reply SNIPS/snip-75.md:56]: yes. but the members shouldn't be reelected by the foundation, so last line should be changed

[the remark this answers is not in the document]
Comment (Eikix) [reply SNIPS/snip-75.md:56]: > yes. but the members shouldn't be reelected by the foundation, so last line should be changed

how do they get reelected?

[the remark this answers is not in the document]
Comment (lucadonnoh) [reply SNIPS/snip-75.md:56]: as said in the first comment, either via an onchain election mechanism like Arbitrum or the SC updates itself

[the remark this answers is not in the document]
Comment (Eikix) [reply SNIPS/snip-75.md]: Ok changing

File: SNIPS/snip-75.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
+---
+snip: 75
+title: Starknet Security Council (SSC)
+author: Elias Tazartes <@Eikix>, Mattéo Georges <@mattegoat>, Patrícia Gil de Brolezzi <@pat_zip>, Luca Donno <@lucadonnoh>
```
Comment (Eikix) [inline SNIPS/snip-75.md:4]: Need to add one last author: luiz-lvj <lvj253@gmail.com>
Waiting to get his full name

**Comment by Eikix** [comment]
Could we get Starkware's opinion and more clarity on the roadmap, @iliav-starkware? (Can't ping Boaz, though I know he owns this track)

**Comment by leo-starkware** [comment]
Hey @Eikix, could you add sections "Rationale" and "Security considerations" to the SNIP, in order to keep a uniform syntax among SNIPs? Let's merge afterwards (after renumbering the SNIP)

**Comment by Eikix** [comment]
> Hey @Eikix, could you add sections "Rationale" and "Security considerations" to the SNIP, in order to keep a uniform syntax among SNIPs? Let's merge afterwards (after renumbering the SNIP)

before merging, could we get Boaz's input or whoever is in charge of the security council is Starknet?
we could get the views of core devs of Starknet, since Security Council triggers downstream big changes to the L1 contracts of Starknet

**Comment by Eikix** [comment]
> There hasn't been any activity on this pull request recently, and in order to prioritize active work, it has been marked as stale. This PR will be closed and locked in 7 days if no further activity occurs. Thank you for your contributions!

Deactivating Stale Label.

I've heard Starkware (SW) and Starknet Foundation (SNF) have iterated and found an alternative design for the Security Council.
Can someone from SW or SNF modify this Pull Request and directly add their design here?

We'll then be able to merge the SNIP and move forward

**Comment by leo-starkware** [comment]
@BoazStark could you address @Eikix 's comment above? I'm not aware of the developments

**Comment by nconsigny** [comment]
> I've heard Starkware (SW) and Starknet Foundation (SNF) have iterated and found an alternative design for the Security Council. Can someone from SW or SNF modify this Pull Request and directly add their design here?

I've helped the Starknet Foundation with this one and I merged the design with the existing proposal, let me know what you think @Eikix

File: SNIPS/snip-75.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
+---
+snip: 75
+title: Starknet Security Council (SSC)
+author: Elias Tazartes <@Eikix>, Mattéo Georges <@mattegoat>, Patrícia Gil de Brolezzi <@pat_zip>, luiz-lvj <lvj253@gmail.com>, Luca Donno <@lucadonnoh>, Nicolas Consigny <@nconsigny>, et al.
```
Comment (Eikix) [inline SNIPS/snip-75.md:4]: main author should be whoever designed the document from SW / SNF since I was not involved

File: SNIPS/snip-75.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+### Eligibility
+Members MUST meet the following criteria:
+- **Technical Competency**: Familiarity with Starknet and secure key management.
+- **Reputation and Alignment**: Trustworthy individuals aligned with Starknet’s vision and values.
+- **Diversity**: Geographic and organizational diversity to prevent overconcentration, with less than 50% from any one country, and fewer than four people from the same organization.
+- **Screening**: All members must pass KYC/AML checks and sign a standard contract accepting their responsibilities within the role.
+
+### Compensation
+Council members receive a monthly stipend paid in STRK tokens for their contributions.
```
Comment (Eikix) [inline SNIPS/snip-75.md:101]: Where is this amount decided? Based on what? Is it public?

File: SNIPS/snip-75.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+- **Ethical Behavior**: Uphold high standards in all interactions.
+- **Positive Communication**: Foster a respectful culture.
+- **Respect and Non-Discrimination**: Promote diversity and inclusivity.
+- **Non-Violence**: Avoid promoting violence.
+- **No Abuse of Position**: Avoid exploiting their position.
+- **Conflict of Interests**: Declare and manage conflicts of interest.
+- **Confidentiality**: Protect sensitive information.
+
+### Adding and Removing Members
+The Starknet Foundation is responsible for appointing and compensating Council members from an administrative perspective. Members may be removed based on the following criteria:
```
Comment (Eikix) [inline SNIPS/snip-75.md:145]: removed by whom? how?

File: SNIPS/snip-75.md [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+- L1 reward supplier
+
+Additionally, the SC will act as the:
+- **Security Admin on the L2 Staking contract**:
+  - Setting the security agent (the entity that can pause the contracts)
+  - Unpausing the contracts
+- **Security Admin on L1 STRK mint manager**:
+  - Setting the security agent (the entity that can revoke allowance)
+- **Security Agent on L1**:
+  - Revoking allowance in case of emergency
```
Comment (Eikix) [inline SNIPS/snip-75.md:176]: revoking allowance of what contract

**Review by Eikix — COMMENTED** [review]
There is no mention of an actor being able to pause Starknet. Is this on purpose?

File: SNIPS/snip-75.md (hunk 4 above)
Comment (nconsigny) [reply SNIPS/snip-75.md:101]: An amount was decided it should be public indeed I just didn't feel like it belonged to the PR 😅

File: SNIPS/snip-75.md (hunk 3 above)
Comment (nconsigny) [reply SNIPS/snip-75.md:4]: I merged both document trying to keep your doc as a basis

**Comment by nconsigny** [comment]
@Eikix updated with the poause flow and a few modifications, wdyt ?

File: SNIPS/snip-75.md [hunk 7]
```diff
@@ -1,7 +1,7 @@
 ---
 snip: 75
 title: Starknet Security Council (SSC)
-author: Elias Tazartes <@Eikix>, Mattéo Georges <@mattegoat>, Patrícia Gil de Brolezzi <@pat_zip>, luiz-lvj <lvj253@gmail.com>, Luca Donno <@lucadonnoh>, Nicolas Consigny <@nconsigny>, et al.
+author: Elias Tazartes <@Eikix>, Mattéo Georges <@mattegoat>, Patrícia Gil de Brolezzi <@pat_zip>, luiz-lvj <lvj253@gmail.com>, Luca Donno <@lucadonnoh>, Nicolas Consigny <@nconsigny>, Henri Lieutaud <@l-henri>, Boaz <@BoazStark>
```
Comment (Eikix) [inline SNIPS/snip-75.md:4]: you can set Henri and Boaz as first writers, wdyt

File: SNIPS/snip-75.md [hunk 8]
```diff
@@ -65,6 +65,24 @@ This is an emergency flow, initiated by the security council; therefore, it is i
...
+### Pausing Flow
+
+Each pausing flow will include two functions:
+
+- **Pause**: Stopping the affected smart contract from further changes of state.
+- **Unpause**: Resuming normal operation.
+
+#### Flow 1: Pausing the Contract
+
+In case of critical incidents, a monitoring entity has the ability to pause the affected contracts immediately. This pause mechanism serves as a first line of defense to contain potential losses while the Security Council deliberates on a potential upgrade.
```
Comment (Eikix) [inline SNIPS/snip-75.md:77]: Does the pause inherently have an expiration?
Like pause until unpause or at most for 1 day

**Comment by Eikix** [comment]
Overall good! what are the next steps wrt SNIP?

File: SNIPS/snip-75.md (hunk 8 above)
Comment (l-henri) [reply SNIPS/snip-75.md:75]: There is no automatic unpause in the current design

File: SNIPS/snip-75.md (hunk 6 above)
Comment (l-henri) [reply SNIPS/snip-75.md:199]: The security agent is the entity that can pause the affected contract. 
See the pausing flow https://github.com/Eikix/SNIPs/blob/security-council/SNIPS/snip-75.md#pausing-flow

**Comment by Eikix** [comment]
changed number to 25 as mentioned by @leo-starkware

File: SNIPS/snip-25.md [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+title: Starknet Security Council (SSC)
+author: Henri Lieutaud <@l-henri>, Boaz <@BoazStark>, Elias Tazartes <@Eikix>, Mattéo Georges <@mattegoat>, Patrícia Gil de Brolezzi <@pat_zip>, luiz-lvj <lvj253@gmail.com>, Luca Donno <@lucadonnoh>, Nicolas Consigny <@nconsigny>
+status: Review
+type: Meta
+created: 2024-07-06
+---
+
+## Abstract
+
+This Meta SNIP lays out the structure, motivation and decisions of the Starknet Security Council. It describes the process and rules that surronds the Starknet Security Council.
```
Comment (leo-starkware) [inline SNIPS/snip-25.md:12]: surround

File: SNIPS/snip-25.md [hunk 10]
```diff
@@ -0,0 +1,267 @@
...
+
+The Starknet Security Council is made up of 12 members who are both geographically and organizationally diverse. They are tasked with safeguarding the security of Starknet. The Council assesses proposed upgrades to the network’s core contracts and approves them if no security risks are identified. In the event of an emergency, the Council has the authority to expedite necessary actions.
+
+## Motivation
```
Comment (leo-starkware) [inline SNIPS/snip-25.md:16]: The whole line 14 appears in section 'Motivation'. Could you delete it from the abstract?

**Review by leo-starkware — CHANGES_REQUESTED** [review]

File: SNIPS/snip-25.md (hunk 9 above)
Comment (l-henri) [reply SNIPS/snip-25.md]: Done!

File: SNIPS/snip-25.md (hunk 10 above)
Comment (l-henri) [reply SNIPS/snip-25.md:14]: Done!

**Review by l-henri — APPROVED** [review]
I think it's good to go!

**Pull request merged**